### DISCUSSION TITLE
AWS ECS plugin stops working with dagster_aws  from 0.22.14 to 0.23.14 :(

### DISCUSSION THREAD

**Discussion by whillas-yabble** [dagster-io/dagster#23162] [Q&A] [answered]
So we bumped the Dagster version and the libraries today from from 1.6.14 to 1.7.14 and the aws lib from  0.22.14 to 0.23.14. We'd had no problems with the ECS adaptor for years and now when trying to run a job we get this ENGINE_EVENT
```
ValueError: list.remove(x): x not in list
  File "/usr/local/lib/python3.10/site-packages/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py", line 383, in _dequeue_run
    instance.run_launcher.launch_run(LaunchRunContext(dagster_run=run, workspace=workspace))
  File "/usr/local/lib/python3.10/site-packages/dagster_aws/ecs/launcher.py", line 394, in launch_run
    run_task_kwargs = self._run_task_kwargs(run, image, container_context)
  File "/usr/local/lib/python3.10/site-packages/dagster_aws/ecs/launcher.py", line 620, in _run_task_kwargs
    task_definition_dict = get_task_definition_dict_from_current_task(
  File "/usr/local/lib/python3.10/site-packages/dagster_aws/ecs/tasks.py", line 293, in get_task_definition_dict_from_current_task
    container_definitions.remove(container_definition)
```
Has there been some other change in the `dagster_aws` module that I should be aware of? Trawling through the dagster change logs to no avail...

**Comment by gibsondan** [comment, marked as answer]
hi @whillas-yabble - thanks for reporting this, I looked into this and I think it may blame to https://github.com/dagster-io/dagster/pull/20663 which went out in 1.7.0. Working on a fix!

**Reply by gibsondan** [reply]
I was able to reproduce the problem and https://github.com/dagster-io/dagster/pull/23166 should fix this, we should be able to get that out in our next dagster-aws release later this week.

**Reply by whillas-yabble** [reply]
Thanks man, thats great news! Thanks for being so on to it :)

**Comment by gibsondan** [comment]
(I think it will only happen if you have include_sidecars set to True - I assume you set that intentionally, but if you don't need it, setting it to False would likely resolve this until the fix is live).

**Reply by whillas-yabble** [reply]
Where is this set? Do you mean in the task definition?

**Reply by gibsondan** [reply]
this would be in your dagster.yaml where you configure the EcsRunLauncher. This bug should only be happening if you have "include_sidecars: true" set in the configuration there (it defaults to false)

**Reply by whillas-yabble** [reply]
So we do! (Not even sure what it means...?)
